pkg/cgroups/cgroupcontrol.go

```diff
@@ -0,0 +1,244 @@
+// Copyright 2020-2021 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cgroups
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"syscall"
+)
+
+// Controller is our enumerated type for cgroup controllers.
+type Controller int
+
+// Group represents a control group.
+type Group string
+
+//nolint
+const (
+	// UnkownController represents a controller of unknown type.
+	UnknownController Controller = iota
+	// blkio cgroup controller.
+	Blkio
+	// cpu cgroup controller.
+	Cpu
+	// cpuacct cgroup controller.
+	Cpuacct
+	// cpuset cgroup controller.
+	Cpuset
+	// devices cgroup controller.
+	Devices
+	// freezer cgroup controller.
+	Freezer
+	// hugetlb cgroup controller.
+	Hugetlb
+	// memory cgroup controller.
+	Memory
+	// net_cls cgroup controller.
+	NetCls
+	// net_prio cgroup controller.
+	NetPrio
+	// per_event cgroup controller.
+	PerfEvent
+	// pids cgroup controller.
+	Pids
+)
+
+var (
+	// controllerNames maps controllers to names/relative paths.
+	controllerNames = map[Controller]string{
+		Blkio:     "blkio",
+		Cpu:       "cpu",
+		Cpuacct:   "cpuacct",
+		Cpuset:    "cpuset",
+		Devices:   "devices",
+		Freezer:   "freezer",
+		Hugetlb:   "hugetlb",
+		Memory:    "memory",
+		NetCls:    "net_cls",
+		NetPrio:   "net_prio",
+		PerfEvent: "perf_event",
+		Pids:      "pids",
+	}
+
+	// controllerNames maps controllers to names/relative paths.
```

Comment:
`// controllerDirs...`
Maybe this could be initialized from controllerNames...
```
for controller, name := range controllerNames {
    controllerDirs[name] = controller
}
```

Reply:
Definitely.